src/dwn-proxy.ts

```diff
@@ -28,8 +39,32 @@ export class DwnProxy {
   dwn: Dwn
   #handlers: Array<IMatchHandler> = []
 
-  constructor(options: DwnProxyOptions) {
+  private constructor(options: DwnProxyOptions) {
     this.options = options
+    if (this.options.dwn?.instance)
+      this.dwn = this.options.dwn.instance
+    this.client = new DwnHttpClient()
+  }
+
+  static async create(options: DwnProxyOptions): Promise<DwnProxy> {
```

Comment:
Decided to follow this pattern as it's what we're doing in lots of our other ts projects... plus it makes testing easier 😸